tests/testscript.py

```diff
@@ -921,6 +921,81 @@ def migration_finished():
 
         computeVM.succeed("virsh list | grep testvm | grep running")
 
+    def test_disk_resize(self):
+        """
+        Test disk resizing during VM runtime.
+
+        Here we test that we can grow and shrink a RAW image. Further, we test
+        that both size modes (KiB and Byte) are working correctly.
+        """
+        # Using define + start creates a "persistent" domain rather than a transient
+        controllerVM.succeed("virsh define /etc/domain-chv.xml")
+        controllerVM.succeed("virsh start testvm")
+
+        disk_size_bytes_10M = 1024 * 1024 * 10
+        disk_size_bytes_100M = 1024 * 1024 * 100
+        disk_size_bytes_200M = 1024 * 1024 * 200
+
+        assert wait_for_ssh(controllerVM)
+
+        controllerVM.succeed("qemu-img create -f raw /tmp/disk.img 100M")
+        controllerVM.succeed(
+            "virsh attach-disk --domain testvm --target vdb --persistent --source /tmp/disk.img"
+        )
+        status, disk_size_guest = ssh(controllerVM, "lsblk --raw -b /dev/vdb | awk '{print $4}' | tail -n1")
+        status, disk_size_host = controllerVM.execute("ls /tmp/disk.img -l | awk '{print $5}'")
+
+        assert(status == 0)
+        assert(int(disk_size_guest) == disk_size_bytes_100M)
+        assert(int(disk_size_host) == disk_size_bytes_100M)
+
+        # Use full file path instead of virtual device name here because both should work with --path
+        controllerVM.succeed(
+            f"virsh blockresize --domain testvm --path /tmp/disk.img --size {disk_size_bytes_10M // 1024}"
+        )
+
+        status, disk_size_guest = ssh(controllerVM, "lsblk --raw -b /dev/vdb | awk '{print $4}' | tail -n1")
+        status, disk_size_host = controllerVM.execute("ls /tmp/disk.img -l | awk '{print $5}'")
+
+        assert(status == 0)
+        assert(int(disk_size_guest) == disk_size_bytes_10M)
+        assert(int(disk_size_host) == disk_size_bytes_10M)
+
+        # Use virtual device name as --path
+        controllerVM.succeed(
+            f"virsh blockresize --domain testvm --path vdb --size {disk_size_bytes_200M // 1024}"
+        )
+
+        status, disk_size_guest = ssh(controllerVM, "lsblk --raw -b /dev/vdb | awk '{print $4}' | tail -n1")
+        status, disk_size_host = controllerVM.execute("ls /tmp/disk.img -l | awk '{print $5}'")
+
+        assert(status == 0)
```

Comment:
nit: in all the cases you only check for the second command's status. the first status is always just overwritten.

you can make the second call a `disk_size_host = controllerVM.succeed("ls /tmp/disk.img -l | awk '{print $5}'")`

`succeed` already checks the status for you and throws an exception otherwise